src/components/App.tsx

```diff
@@ -316,6 +318,11 @@ class App extends React.Component<ExcalidrawProps, AppState> {
         this.setState(
           (state) => ({
             ...actionResult.appState,
+            // use prev library to guard against clearing it when restoring etc.
```

Comment:
i think this breaks the right click "add to library" menu option

Reply:
Just curious, why use `appState` at all for storing the drawing library?

Reply:
If you are worried about getting updates when `localStorage` is updated, you can use the `subscribe` pattern like so:
https://github.com/excalidraw/excalidraw/pull/1834/files#diff-ae70f4fb5856a0e21ead44ca69f0cd7c

The only thing that's missing from that file is to listen to [`storage` events](https://developer.mozilla.org/en-US/docs/Web/API/Window/storage_event) from the `window` and invoke `notify` on each storage event (i.e. `localStorage` updates from other tabs)

Reply:
I agree that we should factor it outside of appState, but let's leave it for a following PR. I'll take a look at the `addToLibrary` action (facepalm on the comment saying we currently don't update library via `syncActionResult`).

Reply:
Now that I look at it, we may wanna migrate to that workflow right now, so as not pile more on hacks still.